frontend/src/static/js/v_zoom.js

```diff
@@ -10,6 +14,22 @@ window.vZoom = {
     getSliceLink(slice) {
       return `${window.getBaseLink(this.info.user.repoId)}/commit/${slice.hash}`;
     },
+
+    updateUser(repos) {
+      const userOrig = repos[this.info.repoIndex][this.info.userIndex];
```

Comment:
The way you implement it indeed works, however, it is not ideal. Instead of this line to get `userOrig`, can you find another way to get this and update this? When updating the data, the original user does not need an index to do so.  You should investigate the whole logic of how v_zoom works before make it also works for `week`. There should be a more elegant method instead of `updateUser`. This time, I will not give you the hint directly. I hope you can find it out by yourself. This is a good opportunity for you to improve your programming skills and debug ability.

Reply:
Feel free to ask me for a hint if you cannot find a better way.

Reply:
> Feel free to ask me for a hint if you cannot find a better way.

Maybe you can hold on to refactor your code. I believe I need to reconsider my way.  Sorry for the disturbance and I will let you know if there are updates.

Reply:
I will try to find a more elegant solution to this as well.

Reply:
@fzdy1914 I am unable to find a more elegant way of updating the user as the splitting of commits based on granularity is done in getFiltered() in v_summary.js, so the only way that I can think of to update the user is to pass the entire filtered repositories and use (repo index, user index) to locate the specific user.

Reply:
I am considering to put these info in the url(which we already have), so when updating, we can use the info in the url directly to update zoom-in tab. This will also benefit remembering the state of zoom-in in url. Now we only need `zoom_start_time` and `zoom_end_time` in url more to remember the zoom. But it is only an idea, you may consider whether it is good or not. Your original implement is also acceptable. I will give a review if you think your original implement is better.

Reply:
For now I guess we can just stick with this current implementation? Only when we are targeting the issue of remembering zoom-in state, we just put the extra info required `(repoIndex, userIndex, zoomStart, zoomEnd)` in the URL instead of passing them as props.